osgbuild/utils.py

```diff
@@ -462,16 +462,24 @@ def popd():
 
 
 def get_local_machine_dver():
-    "Return the distro version (e.g. 'el6', 'el7') of the local machine or None"
+    # type: () -> str
+    """Return the distro version (e.g. 'el6', 'el7') of the local machine
+    or the empty string if we can't figure it out."""
     try:
         redhat_release_contents = slurp("/etc/redhat-release")
-    except EnvironmentError: # some error reading the file
-        return
+        if not redhat_release_contents:
+            return ""  # empty file?
+    except EnvironmentError:  # some error reading the file
+        return ""
 
     for rhellike in ["Scientific", "Red Hat Enterprise", "CentOS"]:
-        try:
-            if rhellike in redhat_release_contents:
-                match = re.search(r"release (\d+)", redhat_release_contents)
+        if rhellike in redhat_release_contents:
+            match = re.search(r"release (\d+)", redhat_release_contents)
+            if match:
                 return "el" + match.group(1)
```

Comment:
```suggestion
                return int(match.group(1))
```

What would you think of just returning an int here instead?  Eventually `"el10" < "el8"` ...